### ISSUE TITLE
WLED Effects are not using color order from outputs.

### ISSUE THREAD

**Issue by CaffeinatedFunctionality** [FalconChristmas/fpp#1947]
Steps to produce: 

1. Use a non RGB order light and set it to it's color order(BRG is what I am using)
2. Run wled effect in scheduler using command
3. Notice the RGB values are not being respected

**Comment by CaffeinatedFunctionality** [comment]
Closing because I think this is user error on my part. If I notice it again I will reopen it.

**Comment by CaffeinatedFunctionality** [comment]
Reopening after a lot of testing and confirmation from other users. When using the Overlay Model Effect, specifically with WLED effects, color order is not being respected. The other effects, bars, blink, color fade, and text, all use the correct color order. It's only the WLED ones that do not. I'm not sure if I can get to this one soon.

**Comment by CaffeinatedFunctionality** [comment]
Closing. I think I misunderstood something

**Comment by coolguymatt** [comment]
Can you explain what you misunderstood? I feel like I have the same issue.

**Comment by CaffeinatedFunctionality** [comment]
Yeah it's still happening. It seems like it doesn't always do it though. It's kind of annoying. I haven't had a chance to research deep into the issue though but it should be reopened. Sorry to keep doing this.

**Comment by CaffeinatedFunctionality** [comment]
This is all about WLED no Xlights. Run WLED with a string with a color order that is not RGB. It's not following that color order. It makes it really hard to use the WLED commands and plugins because this isn't being followed in FPP. 

![Screenshot_20241013_092708_Chrome](https://github.com/user-attachments/assets/15066c89-9b5c-481d-b59e-f9db07f110e7)
![20241013_092718](https://github.com/user-attachments/assets/8e9b34d6-a8fd-448c-881c-33ca5ac98aa3)

**Comment by chassy33** [comment]
I am also running into this issue. I am using Wally's WRGB floods. In xLights I set the color order and upload to controller. All my sequences use this config and no issues. But when using WLED effects in FPP the colors selected don't match what the floods show.

**Comment by dkulp** [comment]
I'm fairly certain it's more of an issue with 4 channel than the order.   I don't believe the PixelOverlayModel things handle anything other than 3 channel pixels at this point.   There are areas in the code like "int c = (y * width * 3) + x * 3;" which is definitely tied to 3 channel.

**Comment by CaffeinatedFunctionality** [comment]
Mine is a three Chanel with this issue. bgr order.

**Comment by ptr727** [comment]
Following, noticed in 8.5.1 while testing upcoming St Patrick's day schedule, the palette was using the wrong color, e.g. Aurora and Forest are green only, but I get reds. Per this report I double checked the color output, and per testing fade red/green/blue output is using correct ordering.

**Comment by CaffeinatedFunctionality** [comment]
So mine is specifically a 3 channel, however it's BGR. It's GS2808 with
dual data, but that's the only other difference.

**Comment by ptr727** [comment]
Mine is 3-channel RBG and color is wrong.

**Comment by darylc** [comment]
@CaffeinatedFunctionality @ptr727 Did the new WLED code Dan committed to master recently help with this?

**Comment by darylc** [comment]
@ptr727 @CaffeinatedFunctionality ?

**Comment by CaffeinatedFunctionality** [comment]
I apologize. I will update my FPP and take a look soon

**Comment by Icesick06** [comment]
I have this issue as well. I have my models uploaded with their correct string properties and im sure it's just something to do with RGBW strings but just the white balance. If I model an effect sequence, the colors are correct when played back. If I just want to use an FPP command with model overlay for a WLED effect, the lights will use a mixed RGB white instead of the standard white channel that string model is set up for in outputs.

**Comment by ptr727** [comment]
Ok, is there a specific version I should wait for to test this, or another way to know if it is ready to be tested again?

**Comment by darylc** [comment]
@ptr727 Keep an eye out for RGBW support in the release notes for future version.  It is not known which version the work will be done in.
